incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -4581,13 +4580,18 @@ private void decode(
                     progress = passwordRO.limit();
                 }
             }
+            return progress;
         }
 
         private void decode(
             MqttPropertiesFW properties)
         {
             willUserPropertiesRW.wrap(willUserPropertiesBuffer, 0, willUserPropertiesBuffer.capacity());
 
+            if (properties == null || properties.length() == 0)
+            {
+                return;
+            }
```

Comment:
Why is this no longer protected at the call site?

Also, please try to avoid multiple returns in the same method, we've avoided that in most of the codebase.